moveit_planners/ompl/ompl_interface/CMakeLists.txt

```diff
@@ -38,17 +38,17 @@ ament_target_dependencies(
   pluginlib
   tf2_eigen
   tf2_ros
-  OMPL
   Boost)
 set_target_properties(
   moveit_ompl_interface PROPERTIES COMPILE_FLAGS
                                    "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 set_target_properties(moveit_ompl_interface PROPERTIES LINK_FLAGS
                                                        "${OpenMP_CXX_FLAGS}")
-
+target_link_libraries(moveit_ompl_interface ompl::ompl)
```

Comment:
Make sure to specify PUBLIC, INTERFACE, or PRIVATE when linking on library targets so it's clear whether this linkage is internal, for external users, or both.

Reply:
Because we call `ament_target_dependencies` on `moveit_ompl_interface` we can't set this level again otherwise you get build errors.
```
CMake Error at ompl_interface/CMakeLists.txt:47 (target_link_libraries):
  The plain signature for target_link_libraries has already been used with
  the target "moveit_ompl_interface".  All uses of target_link_libraries with
  a target must be either all-keyword or all-plain.

  The uses of the plain signature are here:

   * /opt/ros/jazzy/share/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake:146 (target_link_libraries)
   * /opt/ros/jazzy/share/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake:152 (target_link_libraries)
```

Reply:
If I set the level for both `ament_target_dependencies` and `target_link_libraries` it works.

Reply:
Yep! That's what I would do.